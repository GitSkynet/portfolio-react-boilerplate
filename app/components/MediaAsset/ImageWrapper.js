import ResponsiveImage from '../ResponsiveImage';

const ImageWrapper = ResponsiveImage.extend`
  width: 13.75rem;
  border-radius: .5rem;

  > img {
    border-radius: .5rem;
  }
`;

export default ImageWrapper;
