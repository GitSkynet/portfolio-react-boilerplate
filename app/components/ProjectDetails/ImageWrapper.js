import styled from 'styled-components';
import PropTypes from 'prop-types';
import { BASE_LINE_HEIGHT } from '../../styles/typography';

export function renderMediaBreakpoints({ breakpoints = [], imageDimensions = [] }) {
  if (breakpoints.length !== imageDimensions.length) {
    throw new Error(`Mismatch between the length of the breakpoints array, ${breakpoints.length} items, and the imageDimensions array, ${imageDimensions.length} items)`);
  }

  return breakpoints.map((_breakpoint, idx) => {
    const { media: { breakpoint, size } } = _breakpoint;
    const { height, width } = imageDimensions[idx];

    return `@media (${breakpoint}: ${size}) {
      height: ${height}px;
      width: ${width}px;
    };
    `;
  });
}

const ImageWrapper = styled.div`
  margin: ${BASE_LINE_HEIGHT * 3}rem 0;
  text-align: center;

  ${(props) => renderMediaBreakpoints(props)}
`;

ImageWrapper.propTypes = {
  breakpoints: PropTypes.arrayOf(
    PropTypes.shape({
      media: PropTypes.shape({
        breakpoint: PropTypes.string,
        size: PropTypes.string,
      }),
      srcSet: PropTypes.string,
    })
  ),
  imageDimensions: PropTypes.array,
};

export default ImageWrapper;
