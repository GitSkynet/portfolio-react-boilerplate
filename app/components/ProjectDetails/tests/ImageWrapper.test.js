import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import ImageWrapper from '../ImageWrapper';

describe('<ImageWrapper />', () => {
  it('should render and match the snapshot', () => {
    const renderedComponent = shallow(<ImageWrapper />);
    expect(toJson(renderedComponent)).toMatchSnapshot();
  });
});
