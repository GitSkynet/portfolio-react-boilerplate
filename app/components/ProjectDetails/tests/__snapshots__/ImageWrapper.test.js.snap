// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageWrapper /> should render and match the snapshot 1`] = `
.c0 {
  margin: 4.5rem 0;
  text-align: center;
}

@media (max-width:424px) {
  .c0 {
    height: 90px;
    width: 160px;
  }
}

@media (max-width:767px) {
  .c0 {
    height: 900px;
    width: 1600px;
  }
}

<div
  className="c0"
/>
`;

exports[`<ImageWrapper /> should render without any props given 1`] = `
.c0 {
  margin: 4.5rem 0;
  text-align: center;
}

<div
  className="c0"
/>
`;
